### DISCUSSION TITLE
Upgrading from 3.1.1 to 3.4

### DISCUSSION THREAD

**Discussion by muthurathnamswamy-8451** [Kong/kong#13785] [Help] [answered]
Has anyone done upgrade from 3.1.1 to 3.4?

Kong upgrade path recommendation at https://docs.konghq.com/gateway/3.4.x/upgrade/#preparation-review-upgrade-paths:~:text=to%203.4.x.-,3.1.0.x%2D3.1.1.2,-Hybrid suggest the following upgrade path:
1. 3.1.1 - > 3.1.1.3
2. 3.1.1.3 - > 3.2.x
3. 3.2.x - > 3.3.x
4. 3.3.x -> 3.4.x

However there is no 3.1.1.3 release at all for OSS https://cloudsmith.io/~kong/repos/gateway-31/packages/?q=format%3Arpm

Has anyone upgraded from 3.1.1 to 3.4 successfully?  If yes, would appreciate if you could share the upgrade path chosen.

**Comment by Oyami-Srk** [comment, marked as answer]
I believe you can upgrade to 3.2.x and then follow up to 3.4. The Hybrid deploy topology is somehow missing for the OSS version I think. Remember to back up your data before upgrading.

**Comment by muthurathnamswamy-8451** [comment]
I was thinking of the same, thanks for confirming.
